teuthology/openstack/__init__.py

```diff
@@ -830,17 +830,21 @@ def create_security_group(self):
             return
         except subprocess.CalledProcessError:
             pass
-        # TODO(loic): this leaves the teuthology vm very exposed
-        # it would be better to be very liberal for 192.168.0.0/16
-        # and 172.16.0.0/12 and 10.0.0.0/8 and only allow 80/8081/22
-        # for the rest.
+        # TODO(jan): is there a way teuthology can exist but teuthology worker
+        # doesn't? if so the detection logic needs an update.
```

Comment:
Are you referring to VMs or security groups here? We can't assume that workers (target VMs) will always exist, can we? Where is the detection logic you are referring to?